### PR TITLE
Update high level multiplayer docs for Godot 3.0

### PR DESCRIPTION
**Pull request by mhilbrunner**

### CODE REVIEW

**Comment by Zephilinox** [comment]
Does an ID of 0 have a particular meaning, e.g invalid? If you're going to specifically mention the server ID is 1 then it could be worth doing the same for 0.

**Comment by mhilbrunner** [comment]
I'd assume that every ID below 1 would be invalid, yes. To be sure I'd have to dig into the code, however... maybe someone else knows :)

**Comment by mhilbrunner** [comment]
Will update this with more changes to update the docs to Godot 3.0 beta as far as I can.

**Comment by mhilbrunner** [comment]
Updated, all sample code and descriptions should now generally be compatible with Godot 3.0 beta.
Most changes were because of the change from `_set_network_mode()` to `set_network_master()`.
Added some additional explanations and (hopefully) improved the writing a little.

Now roast me :) Hope I've got everything RST correct, have to get used to the syntax a little.

File: learning/features/networking/high_level_multiplayer.rst [hunk 1]
```diff
@@ -362,4 +413,5 @@ who is master for that instance will actually get the function.
 This instance, then, calls the `stun` function in the same instances of that same player (but in different peers), and only those which are set as slave,
 making the player look stunned in all the peers (as well as the current, master one).
 
-.. FIXME: Document the sync keyword
+Note that you could also send the stun() message only to a specific player by using rpc_id(<id>, "exploded", bomb_owner).
+This may not make much sense for an area-of-effect case like the bomb, but in other cases, like single target damage.
```
Comment (cbscribe) [inline learning/features/networking/high_level_multiplayer.rst]: Incomplete?

File: learning/features/networking/high_level_multiplayer.rst [hunk 2]
```diff
@@ -215,39 +258,44 @@ In most games, each player will likely have its own scene. Remember that this is
...
             get_node("/root/world/players").add_child(player)
 
         # Tell server (remember, server is always ID=1) that this peer is done pre-configuring
-        rpc_id(1, "done_preconfiguring", get_tree().get_network_unique_id())
+        rpc_id(1, "done_preconfiguring", selfPeerID)
 	
-Synchronized game start
+
+.. note:: Depending on when you execute pre_configure_game(), you may need to change the any calls to add childs
+          to be deferred, as the SceneTree is locked while the scene is being created (e.g. when `_ready` is being called).
```
Comment (cbscribe) [inline learning/features/networking/high_level_multiplayer.rst]: Either "calls to ``add_child()``" or "calls to add children".  Use double backticks around code in RST.

File: learning/features/networking/high_level_multiplayer.rst [hunk 3]
```diff
@@ -215,39 +258,44 @@ In most games, each player will likely have its own scene. Remember that this is
...
+
+Synchronizing game start
 ^^^^^^^^^^^^^^^^^^^^^^^
 
-Setting up players might take different amount of time on every peer due to lag and any large number of reasons.
-To make sure the game will actually start when everyone is ready, pausing the game can be very useful:
+Setting up players might take different amount of time on every peer due to lag, different hardware and any large number of reasons.
```
Comment (cbscribe) [inline learning/features/networking/high_level_multiplayer.rst]: "different hardware, or any number of other reasons."

File: learning/features/networking/high_level_multiplayer.rst [hunk 4]
```diff
@@ -118,8 +161,8 @@ Functions can be called in two fashions:
 - Reliable: the function call will arrive no matter what, but may take longer because it will be re-transmitted in case of failure.
 - Unreliable: if the function call does not arrive, it will not be re-transmitted, but if it arrives it will do it quickly.
 
-In most cases, Reliable is desired. Unreliable is mostly useful when synchronizing objects that move (sync must happen constantly, 
-and if a packet is lost, it's not that bad because a new one will eventually arrive).
+In most cases, Reliable is desired. Unreliable is mostly useful when synchronizing object positions (sync must happen constantly,
```
Comment (cbscribe) [inline learning/features/networking/high_level_multiplayer.rst]: "Reliable" shouldn't be capitalized.

**Comment by mhilbrunner** [comment]
Updated!

**Pull request merged**